src/components/ColorPicker.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, createElement } from "react";
+
+import { Alert } from "./Alert";
+import * as classNames from "classnames";
+import * as Picker from "react-color";
+
+import "../ui/ColorPicker.css";
+
+interface ColorPickerProps {
+    className?: string;
+    color: string;
+    type: PickerType;
+    mode: Mode;
+    disabled: boolean;
+    style?: object;
+    displayColorPicker: boolean;
+    onChange?: Picker.ColorChangeHandler;
+    close?: () => void;
+    alertMessage?: string;
+    onChangeComplete?: Picker.ColorChangeHandler;
+}
+
+export type PickerType = "sketch" | "chrome" | "block" | "github" | "twitter" | "circle" | "hue" |
+    "alpha" | "slider" | "compact" | "material" | "swatches";
+
+export type Mode = "popover" | "input" | "inline";
+
+export class ColorPicker extends Component<ColorPickerProps, {}> {
+    private components: any = {
+        sketch: Picker.SketchPicker,
+        chrome: Picker.ChromePicker,
+        block: Picker.BlockPicker,
+        github: Picker.GithubPicker,
+        twitter: Picker.TwitterPicker,
+        circle: Picker.CirclePicker,
+        hue: Picker.HuePicker,
+        alpha: Picker.AlphaPicker,
+        slider: Picker.SliderPicker,
+        compact: Picker.CompactPicker,
+        material: Picker.MaterialPicker,
+        swatches: Picker.SwatchesPicker
+    };
+
+    render() {
+        const { mode, type } = this.props;
+
+        return createElement("div", {
+            className: classNames(
+                "widget-color-picker",
+                this.props.className
+            ),
+            style: this.props.style
+        },
+            this.props.children,
+            this.props.displayColorPicker || this.props.mode === "inline"
+                ? createElement("div", {
+                    className: classNames(
+                        { "widget-color-picker-popover": (mode !== "inline" && type !== "alpha" && type !== "slider")
+                    })
+                },
+                    this.props.mode !== "inline"
+                        ? createElement("div", {
```

Comment:
Same here.

Reply:
This Element could be on a single line

Reply:
You have so many conditions that control rendering....hope your tests cover them all :)

Reply:
Yea they will cover